src/api/helpers/formatEntityData.ts

```diff
@@ -99,6 +99,82 @@ export function getWorkBasicInfo(work: Record<string, unknown> | null | undefine
 		};
 }
 
+/**
+ * A function to extract structured author credit details from an Edition ORM entity.
+ * @function
+ * @param {object} edition - an edition ORM entity
+ * @returns {object|null} an object containing the author credit data, or null if not available.
+ *
+ * @example
+ * 		const edition = await orm.func.entity.getEntity(orm, 'Edition', bbid, relations);
+ *		filterAuthorCredit(edition);
+ *		/* => {
+ *			"authorCount": 1,
+ *			"id": 135,
+ *			"names": [
+ *				{
+ *					"authorBBID": "e5c4e68b-bfce-4c77-9ca2-0f0a2d4d09f0",
+ *					"authorCreditID": 135,
+ *					"name": "J. K. Rowling"
+ *				}
+ *			]
+ *		}
+ */
+
+export function filterAuthorCredit(edition: any) {
+	if (_.isNil(edition) || _.isNil(edition.authorCredit)) { return null; }
+
+	const credit = edition.authorCredit;
+	return {
+		authorCount: credit.authorCount || null,
+		id: credit.id || null,
+		names: (credit.names || []).map(({authorCreditID, authorBBID, name}) => ({
```

Comment:
Proposed refactoring with destructuring assignment:

```suggestion
	const {authorCredit} = edition;
	const {id, authorCount, names} = authorCredit;
	return {
		authorCount: authorCount ?? null,
		id: id ?? null,
		names: (names ?? []).map(({authorCreditID, authorBBID, name}) => ({
```

I also wonder if we want to return `null` or if we could instead use shorthand properties like you did below in the mapping function
```
return {
		authorCount,
		id,
....
```